height_mapping:
  lidarCloudTopic: "/velodyne/points"
  rgbCloudTopic: "/camera/depth/color/points"
  baselinkFrame: "base_link"
  odometryFrame: "odom"
  mapFrame: "map"
  debugMode: true

  # Pointcloud Preprocessing
  minHeightThreshold: -0.05 # indoor: -0.05, outdoor: -0.2
  maxHeightThreshold: 0.5
  minRangeThreshold: 0.3
  maxRangeThreshold: 10.0
  minDepthThreshold: 1.0
  maxDepthThreshold: 5.0

  # Height Map
  gridResolution: 0.05
  mapLengthX: 15.0
  mapLengthY: 15.0
  heightEstimatorType: "StatMean" # "StatMean", "KalmanFilter", "MovingAverage"

  # Process
  poseUpdateRate: 10
  mapPublishRate: 10

global_mapping:
  # Topics
  globalMapTopic: "/height_mapping/elevation_cloud_global"
  mapRegionTopic: "/height_mapping/map_region_global"

  # Map parameters
  gridResolution: 0.1
  mapLengthXGlobal: 600
  mapLengthYGlobal: 600
  heightEstimatorType: "StatMean" # "StatMean", "MovingAverage", "KalmanFilter"

  # Node settings
  globalMapVisualizationRate: 10.0

  # Map saver parameters
  mapSaveDir: "$(find height_mapping)/map_image"
